org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/nexus/BasicScanTest.java

```diff
@@ -204,8 +202,9 @@ private void checkMetadataScannables(final ScanModel scanModel, NXinstrument ins
 		DataNode dataNode;
 		Dataset dataset;
 
-		Collection<IScannable<?>> perScan  = scanModel.getMonitors().stream().filter(scannable -> scannable.getMonitorRole()==MonitorRole.PER_SCAN).collect(Collectors.toList());
-        for (IScannable<?> metadataScannable : perScan) {
+		if (scanModel.getMonitorsPerScan() == null) return;
+
+        for (IScannable<?> metadataScannable : scanModel.getMonitorsPerScan()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This line will not be executed conditionally; only the first line of this 3-line block will be. The rest will execute unconditionally. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2681)